packages/@react-stately/datepicker/src/useDatePickerState.ts

```diff
@@ -113,8 +113,14 @@ export function useDatePickerState<T extends DateValue = DateValue>(props: DateP
   };
 
   let selectTime = (newValue: TimeValue) => {
-    if (selectedDate && newValue) {
-      commitValue(selectedDate, newValue);
+    if (selectedDate) {
```

Comment:
there are a lot of ways this conditional could be written so i was wondering if there was a particular way we'd like to do this?